addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -346,6 +346,18 @@ var FieldMany2One = AbstractField.extend({
     _getDisplayName: function (value) {
         return value.split('\n')[0];
     },
+    /**
+    * Returns the display_name from a string which contains it but was altered
+    * as a result of the show_vat option.
+    * Note that the split is done on a 'figuredash', not a standard dash.
+    *
+    * @private
+    * @param {string} value
+    * @returns {string} display_name without TaxID
+    */
+    _getDisplayNameWithoutVAT: function (value) {
```

Comment:
I would keep it in `partner_autocomplete` unless you would see another use case where it applies.

Reply:
Technically we have no guarantee that it won't be used elsewhere, but for now it is only in the res_partner_many2one widget, so it can go into partner_autocomplete.